pkg/resources/kafka/kafka.go

```diff
@@ -102,9 +113,42 @@ func New(client client.Client, directClient client.Reader, cluster *v1beta1.Kafk
 			DirectClient: directClient,
 			KafkaCluster: cluster,
 		},
-		kafkaClientProvider: kafkaClientProvider,
+		kafkaClientProvider:        kafkaClientProvider,
+		CruiseControlScalerFactory: scale.ScaleFactoryFn(),
 	}
 }
+
+func getBrokerAzMap(cluster *v1beta1.KafkaCluster) map[int32]string {
```

Comment:
Suggestion: It'd be appreciated if we can add some quick unit tests for func like this one, but since the caller `handleRollingUpgrade` is tested, I am not going to ask for any more changes.

Reply:
Let's do this properly. I added unit tests for it in cae78f54b77050afa23466899d1ee7a38316dee3, please have a look. Thanks!

P.S. I saw you were doing heavy lifting with kraft, so no worries and thanks for the review. :)